views/components/revealable_content.erb

```diff
@@ -0,0 +1,10 @@
+<span class="revealable-content">
+  <span class="shadow-content flex items-center">
+    ●●●●●●
+    <img src="/icons/hero-eye.svg" width="18px" class="reveal-button inline-block cursor-pointer ms1"/>
+  </span>
+  <span class="revealed-content hidden flex items-center">
+    <%== content %>
```

Comment:
There are two ways to render a variable: `<%==` and `<%=`. `<%==` doesn't quote content and renders it as it. `<%=` quote everything, so it prints i.e. html tags as string. So if the content doesn't have any special html tag, it's safer to use `<%=`. Especially for user provided data to prevent any kind of injection.